pkg/agent/manifest.go

```diff
@@ -229,6 +231,16 @@ func agentDeployment(namespace string, agentScope string, opts ManifestOptions,
 	// additional tolerations from cluster
 	dep.Spec.Template.Spec.Tolerations = append(dep.Spec.Template.Spec.Tolerations, opts.AgentTolerations...)
 
+	// overwrite affinity if present on cluster
+	if opts.AgentAffinity != nil {
+		dep.Spec.Template.Spec.Affinity = opts.AgentAffinity
+	}
+
+	// set resources if present on cluster
+	if opts.AgentResources != nil {
+		dep.Spec.Template.Spec.Containers[0].Resources = *opts.AgentResources
```

Comment:
shouldn't we change Resources in all containers? Imagine we have a pod with multiple containers, I think it should be applied to all containers.

Reply:
The agent deployment is only one container. This is in line with the env vars below, they're also added just to the first container.

Reply:
what if someone is using something like Istio that injects a sidecar container?